src/components/Answer.tsx

```diff
@@ -166,21 +150,27 @@ export const Answer = (props: AnswerProps) => {
       <>
         {type === 'teacher-preview' && correctIncorrectIcon}
         {selectedCount}
-        {radioBox}
+        <input
```

Comment:
Rather than inline JSX, it would be more clear to have each body candidate be a named React component.
The `if` block would be concise, and a reader could ignore implementation details unless needed.
And you'd have `<Body />` (or some such) rather than `{body}`.

Reply:
I had wanted to make minimal changes, but I do not mind refactoring this. It turns out there were a few things that were not clear. For example, `selectedCount` was always `undefined` in the version of the body you have highlighted in this comment because it was only being set when `type` was teacher-review.